### ISSUE TITLE
Error: Cannot read property 'loc' of undefined

### ISSUE THREAD

**Issue by azu** [textlint/textlint#661]
https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/129
Probably, This issue is related with textlint's core.

**Comment by azu** [comment]
It is related with https://github.com/textlint/textlint/issues?q=is%3Aissue+is%3Aclosed

```js
        var isPluginReturnAnAST = ast_tester_1.isTxtAST(preProcessResult);
        var textForAST = isPluginReturnAnAST ? text : preProcessResult.text;
        var ast = isPluginReturnAnAST ? preProcessResult : preProcessResult.ast;

```

is not work correctly. 
`isPluginReturnAnAST` is false positive

**Comment by azu** [comment]
https://github.com/azu/textlint-rule-no-dead-link-issues-129
reproduce repository

**Comment by azu** [comment]
```json
{
  "type": "Document",
  "children": [
    {
      "type": "List",
      "ordered": false,
      "start": null,
      "loose": true,
      "children": [
        {
          "type": "ListItem",
          "loose": true,
          "checked": null,
          "children": [
            {
              "type": "Paragraph",
              "children": [
                {
                  "type": "Str",
                  "value": "Below are the two types of parameters expected:",
                  "loc": {
                    "start": {
                      "line": 1,
                      "column": 2
                    },
                    "end": {
                      "line": 1,
                      "column": 49
                    }
                  },
                  "range": [
                    2,
                    49
                  ],
                  "raw": "Below are the two types of parameters expected:"
                }
              ],
              "loc": {
                "start": {
                  "line": 1,
                  "column": 2
                },
                "end": {
                  "line": 1,
                  "column": 49
                }
              },
              "range": [
                2,
                49
              ],
              "raw": "Below are the two types of parameters expected:"
            },
            {
              "type": "CodeBlock",
              "lang": null,
              "value": "\t* Type `String` called `code`\n\t* Type `boolean` called `accepted`",
              "loc": {
                "start": {
                  "line": 3,
                  "column": -1
                },
                "end": {
                  "line": 4,
                  "column": 35
                }
              },
              "range": [
                50,
                117
              ],
              "raw": "\n\t* Type `String` called `code`\n\t* Type `boolean` called `accepted`"
            }
          ],
          "loc": {
            "start": {
              "line": 1,
              "column": 0
            },
            "end": {
              "line": 4,
              "column": 35
            }
          },
          "range": [
            0,
            117
          ],
          "raw": "* Below are the two types of parameters expected:\n\n\t* Type `String` called `code`\n\t* Type `boolean` called `accepted`"
        }
      ],
      "loc": {
        "start": {
          "line": 1,
          "column": 0
        },
        "end": {
          "line": 4,
          "column": 35
        }
      },
      "range": [
        0,
        117
      ],
      "raw": "* Below are the two types of parameters expected:\n\n\t* Type `String` called `code`\n\t* Type `boolean` called `accepted`"
    }
  ],
  "loc": {
    "start": {
      "line": 1,
      "column": 0
    },
    "end": {
      "line": 4,
      "column": 35
    }
  },
  "range": [
    0,
    117
  ],
  "raw": "* Below are the two types of parameters expected:\n\n\t* Type `String` called `code`\n\t* Type `boolean` called `accepted`"
}
```

text
```markdown
* Below are the two types of parameters expected:

	* Type `String` called `code`
	* Type `boolean` called `accepted`
```

**Comment by azu** [comment]
Assertion Error
```
AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert.ok(start.column >= 0)

    at test (/Users/azu/ghq/github.com/azu/textlint-rule-no-dead-link-issues-129/node_modules/@textlint/ast-tester/lib/ast-tester/src/index.js:58:20)
    at Array.forEach (<anonymous>)
    at test (/Users/azu/ghq/github.com/azu/textlint-rule-no-dead-link-issues-129/node_modules/@textlint/ast-tester/lib/ast-tester/src/index.js:35:23)
    at Array.forEach (<anonymous>)
    at test (/Users/azu/ghq/github.com/azu/textlint-rule-no-dead-link-issues-129/node_modules/@textlint/ast-tester/lib/ast-tester/src/index.js:35:23)
    at Array.forEach (<anonymous>)
    at Object.test (/Users/azu/ghq/github.com/azu/textlint-rule-no-dead-link-issues-129/node_modules/@textlint/ast-tester/lib/ast-tester/src/index.js:35:23)
    at TextlintKernel._parallelProcess (/Users/azu/ghq/github.com/azu/textlint-rule-no-dead-link-issues-129/node_modules/@textlint/kernel/lib/kernel/src/textlint-kernel.js:136:48)
    at /Users/azu/ghq/github.com/azu/textlint-rule-no-dead-link-issues-129/node_modules/@textlint/kernel/lib/kernel/src/textlint-kernel.js:86:26
npm ERR! Test failed.  See above for more details.
```

**Comment by azu** [comment]
```json
                        {
                            "type": "CodeBlock",
                            "lang": null,
                            "value": "\t* Type `String` called `code`\n\t* Type `boolean` called `accepted`",
                            "loc": {
                                "start": {
                                    "line": 3,
                                    "column": -1
                                },
                                "end": {
                                    "line": 4,
                                    "column": 35
                                }
                            },
                            "range": [
                                50,
                                117
                            ],
                            "raw": "\n\t* Type `String` called `code`\n\t* Type `boolean` called `accepted`"
                        }
```
This code block is strange.
`-1` ...

**Comment by azu** [comment]
Probably, it is old remarked's bug

**Comment by azu** [comment]
* [Release remark-parse@7.0.0 · remarkjs/remark](https://github.com/remarkjs/remark/releases/tag/remark-parse%407.0.0)
* [Release remark-parse@6.0.0 · remarkjs/remark](https://github.com/remarkjs/remark/releases/tag/remark-parse%406.0.0)

**Comment by lrobledo** [comment]
Hi, I'm still having this issue. I've tried textlint 11.6.3 as well as earlier versions. Here's my .textlintrc file: 

```
{
    "filters": {
        "comments": true
    },
    "plugins": {
        "json": true,
        "html": true,
        "@textlint/markdown": {
            "extensions": [
                ".def"
            ]
        }
    },
    "rules": {
        "terminology": {
            "defaultTerms": true,
        },
        "stop-words": {
            "defaultWords": false,
            "words": "stopWords.txt"
        },
        "spelling": false,
    }
}
```

I've also tried running it with each of those rules on their own and it breaks every time.

Can you please advise??

**Comment by azu** [comment]
Fm,  #662 fixes markdown plugin.
Probably, other plugins has same issue.

Can you create a minimal reproduce repository? We need to debug it.

**Comment by lrobledo** [comment]
Sure I can do that. Should I create a repo and give you the link?

**Comment by azu** [comment]
Yes please if its possible.

**Comment by lrobledo** [comment]
https://github.com/lrobledo/textlint-bug-example/tree/master

Here it is. Seems like the problem is with the JSON plugin. HTML works fine

**Comment by azu** [comment]
Thanks. It look like that json plugin does not define `loc` property.
It is the reason of error.
https://github.com/duck8823/textlint-plugin-json/blob/70de3684fe32fccc4941fe09e7d29feb4a7d21eb/src/JSONProcessor.js#L65
